lib/agentx.c

```diff
@@ -23,17 +23,53 @@
 #include "libfrr.h"
 #include "xref.h"
 #include "lib/libagentx.h"
+#include "ringbuf.h"	 /* for ringbuf_remain, ringbuf_peek, ringbuf_.. */
+#include "frr_pthread.h" /* for struct frr_pthread */
 
 XREF_SETUP();
 
+
+extern int agentx_stop(struct frr_pthread *fpt, void **result);
+
 DEFINE_HOOK(agentx_enabled, (), ());
 
 //bool agentx_enabled = false;
 
 static struct event_loop *agentx_tm;
+static struct event_loop *main_tm;
```

Comment:
can we name this something better?  and at least put some documentation into why agentx code has too event_loop data structure pointers.  Let's help our future selves out here

Reply:
taken ento account